mcp/sensors/star_camera_receive.c

```diff
@@ -0,0 +1,347 @@
+        return NULL;
+    }
+    // Fix all this here
+    while (listening) {
+        listening = check_sc_thread_bool(which_sc);
+        if (first_time) {
+            first_time = 0;
+            if ((returnval = getaddrinfo(NULL, socket_target->port, &hints, &servinfo)) != 0) {
+                blast_err("getaddrinfo: %s\n", gai_strerror(returnval));
+                return NULL;
+            }
+            // loop through all the results and bind to the first we can
+            for (servinfoCheck = servinfo; servinfoCheck != NULL; servinfoCheck = servinfoCheck->ai_next) {
+                // since a success needs both of these, but it is inefficient to do fails twice,
+                // continue skips the second if on a fail
+                if ((sockfd = socket(servinfoCheck->ai_family, servinfoCheck->ai_socktype,
+                        servinfoCheck->ai_protocol)) == -1) {
+                    perror("listener: socket");
+                    continue;
+                }
+                if (bind(sockfd, servinfoCheck->ai_addr, servinfoCheck->ai_addrlen) == -1) {
+                    close(sockfd);
+                    perror("listener: bind");
+                    continue;
+                }
+                break;
+            }
+            if (servinfoCheck == NULL) {
+                blast_err("listener: failed to bind socket\n");
+                return NULL;
+            }
+            // set the read timeout (if there isnt a message) to 100ms
+            struct timeval read_timeout;
+            read_timeout.tv_sec = 0;
+            read_timeout.tv_usec = 100000;
+            setsockopt(sockfd, SOL_SOCKET, SO_RCVTIMEO, &read_timeout, sizeof read_timeout);
+            // now we set up the print statement vars
+            // need to cast the socket address to an INET still address
+            struct sockaddr_in *ipv = (struct sockaddr_in *)servinfo->ai_addr;
+            // then pass the pointer to translation and put it in a string
+            inet_ntop(AF_INET, &(ipv->sin_addr), ipAddr, INET_ADDRSTRLEN);
+            blast_info("Parameter receiving target is: %s\n", socket_target->ipAddr);
+        }
+        numbytes = recvfrom(sockfd, &received_parameters, sizeof(received_parameters)+1 ,
+         0, (struct sockaddr *)&their_addr, &addr_len);
+        // we get an error everytime it times out, but EAGAIN is ok, other ones are bad.
+        if (numbytes == -1) {
+            err = errno;
+            if (err != EAGAIN) {
+                blast_err("Errno is %d\n", err);
+                blast_err("Error is %s\n", strerror(err));
+                perror("Recvfrom");
+            }
+        } else {
+            where_to_unpack(received_parameters, which_sc);
+        }
+        if (check_for_reset(which_sc)) {
+            blast_info("received reset command...");
+            first_time = 1;
+            freeaddrinfo(servinfo);
+            close(sockfd);
+        } else {
+            usleep(200000);
```

Comment:
No magic numbers, give this a name so a maintainer who wants to change it has some idea of why it's there/has the value it has.

Reply:
done